### PR TITLE
Update to MoorDyn v2

### PR DESCRIPTION
**Pull request by RyanDavies19**
This updated the coupling of WECSim and MoorDyn to MoorDyn-C v2 using the MoorDyn v1 API. Additional updates to the MoorDyn libraries and input files will be added separately. 

The only new feature added is the ability to specify the path and file name of the MoorDyn input file. This will be the same path where outputs are written. The figures below show that this update performs as well as the MoorDyn v1 coupling. This update opens the door to future utilization of the MoorDyn v2 features. 

Fairten_comp
float_heave_response
float_surge_response
spar_heave_response
spar_surge_response

### CODE REVIEW

**Comment by jtgrasb** [comment]
Corresponds with [WEC-Sim Applications PR 45 ](https://github.com/WEC-Sim/WEC-Sim_Applications/pull/45) and [WEC-Sim/MoorDyn PR 1](https://github.com/WEC-Sim/MoorDyn/pull/1).

**Review by akeeste — CHANGES_REQUESTED** [review]
@jtgrasb @RyanDavies19 Thanks for your work. It's great to see updates for MoorDyn in WEC-Sim! After installing MinGW and adding ``WEC-Sim/MoorDyn`` to the source, the applications case runs to completion.

A few requested changes:
- [x] Update responseClass.m line 347 to account for new output file naming
   - I received several messages that "No moorDyn *.out file saved for Line1", etc (These should probably be warnings). However there are output files located in ``CASE/Mooring/`` with names ``lines_Line1.out``, etc. It looks like the name of the MoorDyn input file is being appended to all output files. We should account for this. The case still runs when the responseClass is updated
- [x] Update the variable name of ``moorDyn_infile`` to WEC-Sim's camelCase convention. I suggest ``moorDynInputFile``
- [ ] Please briefly note here what changes were made to the Mooring library for our future reference. It appears changes are primarily in the ``Properties/Callbacks`` of ``WECSim_Lib_Moorings/MoorDyn/MoorDyn/MoorDyn``. Anything else of note?

A couple other non-critical comments for future work:
* Where in the MoorDyn repo do the files in ``WEC-Sim/MoorDyn`` come from? As an alternative to copying files to WEC-Sim/MoorDyn and then to the WEC-Sim source, could users point straight to a local MoorDyn clone?
* We should add that information to the developer manual

**Comment by jtgrasb** [comment]
Thanks for the review @akeeste. I updated accordingly. 

The updates to the mooring library are only in the ``Properties/Callbacks`` of ``WECSim_Lib_Moorings/MoorDyn/MoorDyn/MoorDyn``. These updates changed the initFcn, stopFcn, and closeFcn to use the new library name ('libmoordyn') and the new MoorDyn.h header file which serves as a wrapper for MoorDyn.

File: source/objects/mooringClass.m [hunk 1]
```diff
@@ -41,11 +41,13 @@
         moorDynLines (1,1) {mustBeInteger, mustBeNonnegative} = 0           % (`integer`) Number of lines in MoorDyn. Default = ``0``
         moorDynNodes (1,:) {mustBeInteger, mustBeNonnegative} = []          % (`integer`) number of nodes for each line. Default = ``'NOT DEFINED'``
         name (1,:) {mustBeText}                             = 'NOT DEFINED' % (`string`) Name of the mooring. Default = ``'NOT DEFINED'``
+        moorDyn_infile (1,:) {mustBeText}           = 'Mooring/lines.txt'   % (`string`) Name of the MoorDyn input file path. Outputs will be written to this path. Default = ``Mooring/lines.txt``
         lookupTableFlag                                     = 0;            % (`integer`) Flag to indicate a mooring look-up table, 0 or 1. Default = ``0``
         lookupTableFile                                     = '';           % (`string`) Mooring look-up table file name. Default =  ``''``;
     end
 
     properties (SetAccess = 'private', GetAccess = 'public') %internal
+        libname (1,:) {mustBeText} = 'libmoordyn'                         % (`string`) Name of the mooring. Default = ``'NOT DEFINED'``
```
Comment (akeeste) [inline source/objects/mooringClass.m:50]: I don't see this variable used anywhere else. Can we remove it?

**Comment by akeeste** [comment]
Thanks @jtgrasb @RyanDavies19 for the updates and comments. I missed one thing in my review on ``libname``, see above

**Comment by RyanDavies19** [comment]
@akeeste you can remove that libname variable. That was left over from a different approach we were trying, I forgot to remove it.

**Comment by akeeste** [comment]
@RyanDavies19 sounds good. I'll merge this PR and fix it on dev

**Pull request merged**